src/utils.js

```diff
@@ -270,4 +293,135 @@ export async function saveFileAndRetrieveUrl(fileInfo) {
     newPageUrl = category ? `/sites/${siteName}/collections/${category}/${newFileName}` : `/sites/${siteName}/pages/${newFileName}`
   }
   return newPageUrl
-}
\ No newline at end of file
+}
+
+/*
+ * Util functions for generating file identifiers (the numeric/alphanumeric strings which filenames begin with)
+ */
+
+// Generate new filename when creating page in a collection
+const generateNewCollectionFileName = async ({
+  originalThirdNavTitle,
+  thirdNavTitle,
+  collectionPageData,
+  baseApiUrl,
+  title,
+  siteName,
+  category,
+}) => {
+  let newFileName
+
+  // New file name is also dependent on whether the file has been moved into or out of a third nav
+  if (originalThirdNavTitle !== thirdNavTitle && collectionPageData) {
+    // Case: Move from within a third nav section to outside of it within the collection
+    if (!thirdNavTitle) {
+      const groupIdentifier = await generateGroupIdentifier(collectionPageData, false, baseApiUrl)
+      newFileName = generateCollectionPageFileName(title, groupIdentifier);
+
+    } else {
+      // Assumption: third nav titles are unique
+      const thirdNavSection = collectionPageData.filter((section) => section.type === 'third-nav' && section.title === thirdNavTitle)
+
+      let groupIdentifier
+      // Case: Create a new third nav section
+      if (thirdNavSection.length === 0) {
+        groupIdentifier = await generateGroupIdentifier(collectionPageData, true, baseApiUrl, true)
+
+      // Case: Move from outside of third nav into a third nav OR
+      // Case: Move from one third nav section into another third nav section
+      } else {
+        // Move the file to be the last file in the third nav
+        groupIdentifier = await generateGroupIdentifier(thirdNavSection[0].contents, true, baseApiUrl)
+      }
+
+      newFileName = generateCollectionPageFileName(title, groupIdentifier);
+    }
+  } else {
+    // Case: Creating a new page from the workspace and assigning to collection BUT not third nav
+    // Case: Creating a new page from within a collection
+    if (!originalThirdNavTitle && !thirdNavTitle) {
+      const apiUrl = `${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/collections/${category}`
+      const groupIdentifier = await generateGroupIdentifier(null, false, apiUrl)
+      newFileName = generateCollectionPageFileName(title, groupIdentifier);
+
+    // Case: Creating a new page from the workspace and assigning to collection AND third nav
+    } else if (!collectionPageData && thirdNavTitle) {
+      const apiUrl = `${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/collections/${category}/pages`
+      const groupIdentifier = await generateGroupIdentifier(null, true, apiUrl, false, thirdNavTitle)
+      newFileName = generateCollectionPageFileName(title, groupIdentifier);
+    }
```

Comment:
These cases don't cover the generation of new 3rd navs from within a collection/from the workspace

Reply:
Fixed with [fba6ecf](https://github.com/isomerpages/isomercms-frontend/pull/218/commits/fba6ecf9e7b9991448239a69b7746405ab55bf9f)